### PR TITLE
Using Ruby On Rails For Your Back End: Fix link URLs

### PR DESCRIPTION
**Pull request by ayrt-n**
Complete the following REQUIRED checkboxes:
-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`

-   [x] I have previewed all lesson files included in this PR with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure the Markdown content is formatted correctly
-   [x] I have ensured all lesson files included in this PR follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

<hr>

**1. Because:**
I believe the content at the current link (https://edgeguides.rubyonrails.org/working_with_javascript_in_rails.html) has been updated and no longer aligns with the exercise item "Refresh yourself on Rails AJAX from RailsGuide"

I found what I believe to be the original guide (https://guides.rubyonrails.org/v4.2/working_with_javascript_in_rails.html), which has headers that match up with the Knowledge Check Questions ("How does 'unobtrusive JavaScript' work?" and "Why would you want to use AJAX to load large batches of data?")

**2. This PR:**
- Update the exercise link for "Refresh yourself on Rails AJAX from RailsGuides"
- Update the associated Knowledge Check links

**3. Additional Information:**
N/A

### CODE REVIEW

**Comment by ayrt-n** [comment]
Well, I was just digging around rails guide and found a more up-to-date version which may be more appropriate: https://guides.rubyonrails.org/v6.1/working_with_javascript_in_rails.html

Probably better to use this link and I can make the changes if desired.

**Comment by ChargrilledChook** [comment]
@ayrt-n Good catch with this content not matching up. This is releated to some fairly large changes in how Rails 7 handles Javascript in general. Personally I think it would be better to update the lesson to reflect more modern Rails, rather than refer to older document. UJS particularly feels a bit out of place to me in this lesson as it is esentially deprecated.

Ajax I'm a little bit less sure about. Turbo abstracts a lot of of what you would do manually with Ajax / UJS away, but is potentially still a useful thing to know.

Updating the page to use the 6.1 docs in the short term so that this lesson makes sense may be a good short term solution. 

@TheOdinProject/rails-path Thoughts on this?

**Comment by rlmoser99** [comment]
It make sense to me to do a short-term update, and make a new issue to revamp this lesson for Rails 7.

**Comment by ChargrilledChook** [comment]
@rlmoser99 Agreed :+1: 

@ayrt-n Would you like to update the links to use the 6.1 documentation like you suggested? If you feel like going above and beyond, a note about big changes to how Rails 7 handles JavaScript and that these docs are for older behaviour could also be useful.

**Comment by ayrt-n** [comment]
I will update the link and look into adding a note on Rails 7 changes! I'll probably just make the link change quickly and then perhaps look into the note in a separate PR

**Comment by ChargrilledChook** [comment]
@ayrt-n Sounds great, thanks!

File: javascript/javascript_and_the_backend/using_rails_for_your_backend.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ Are you done? Good. Next, it's time to practice allowing your front end JavaScri
 
 1. [Check out "Using JavaScript in your Rails App" from Daniel Kehoe](http://railsapps.github.io/rails-javascript-include-external.html). It is long and covers a lot of ground, but it's got great content. Some of the stuff on dependencies can be skimmed, but pay attention to the `content_for` stuff at the bottom.
 
-2. [Refresh yourself on Rails AJAX from RailsGuides](http://edgeguides.rubyonrails.org/working_with_javascript_in_rails.html) (just skim the top few sections).
+2. [Refresh yourself on Rails AJAX from RailsGuides](https://guides.rubyonrails.org/v4.2/working_with_javascript_in_rails.html) (just skim the top few sections).
```
Comment (ChargrilledChook) [inline javascript/javascript_and_the_backend/using_rails_for_your_backend.md:27]: If we can change this (and the other references) to the 6.1 docs per your suggestion, we can get this one merged :+1:

**Review by ChargrilledChook — CHANGES_REQUESTED** [review]

File: javascript/javascript_and_the_backend/using_rails_for_your_backend.md (hunk 1 above)
Comment (ayrt-n) [reply javascript/javascript_and_the_backend/using_rails_for_your_backend.md:27]: Sorry busy week and I made some git mistakes I was trying to figure out!

Can I just clarify, should I sync this to the upstream again (which will create a commit for a merge request on my branch) in addition to the changes, or just simply update the links?

Comment (ChargrilledChook) [reply javascript/javascript_and_the_backend/using_rails_for_your_backend.md:27]: @ayrt-n No worries at all, I appreciate you taking time to  make the course better :tada: Please feel free to ask any questions if you need, I'm happy to help.

In this case, it should be fine to simply push changes to your own branch on your fork and they'll automatically show up here. If there's conflicts, they'll show up and we can cross that bridge when we come to it (but it's unlikely). 

If you do want to sync with the upstream there would no harm in it, though I'm fairly confident no one eles has been working on this particular file in the meantime.

Comment (ayrt-n) [reply javascript/javascript_and_the_backend/using_rails_for_your_backend.md]: Pushed up the changes, just let me know if there are any issue! @ChargrilledChook

**Review by ChargrilledChook — APPROVED** [review]
Thanks @ayrt-n, looks great!

**Pull request merged**